bindings/c/CONTRIBUTING.md

```diff
@@ -0,0 +1,83 @@
+# Contributing
+1. [Contributing](#contributing)
+   1. [Setup](#setup)
+      1. [Using a dev container environment](#using-a-dev-container-environment)
+      2. [Bring your own toolbox](#bring-your-own-toolbox)
+   2. [Build](#build)
+   3. [Test](#test)
+   4. [Docs](#docs)
+   5. [Misc](#misc)
+
+## Setup
+
+### Using a dev container environment
+OpenDAL provides a pre-configured [dev container](https://containers.dev/) that could be used in [Github Codespaces](https://github.com/features/codespaces), [VSCode](https://code.visualstudio.com/), [JetBrains](https://www.jetbrains.com/remote-development/gateway/), [JuptyerLab](https://jupyterlab.readthedocs.io/en/stable/). Please pick up your favourite runtime environment.
+
+The fastest way is:
+
+[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/apache/incubator-opendal?quickstart=1&machine=standardLinux32gb)
+
+### Bring your own toolbox
+To build OpenDAL C binding, the tools you need are very simple.
+
+The following is all you need
+- **A C++ compiler** that supports **c++14**, *e.g.* clang++ and g++
+- **A Rust toolchain** that satisfies the MSRV
+- **(optional)** if you want to format the code, you need to install **clang-format**
+    - The `opendal.h` is not formatted by hands when you contribute, please do not format the file. **Use `make format` only.**
+    - If your contribution is related to the files under `./tests`, you may format it before submitting your pull request. But notice that different versions of `clang-format` may format the files differently.
+- **(optional)** **GTest(Google Test)** need to be installed if you want to build the BDD (Behavior
+  Driven Development) tests. To see how to build, check [here](https://github.com/google/googletest).
```

Comment:
It's not optional for c binding's developer: they must have them setup so that they can develop c binding.